test/unit/communication/test_reduce.cpp

```diff
@@ -22,241 +24,173 @@ using namespace dlaf::comm;
+  constexpr std::size_t nblocks = 3;
+  constexpr std::size_t block_size = 2;
+  constexpr std::size_t block_distance = 5;
 
-  const std::size_t memory_footprint = (nblocks - 1) * block_distance + block_size;
+  constexpr std::size_t memory_footprint = (nblocks - 1) * block_distance + block_size;
   TypeParam data_strided[memory_footprint];
 
-  const int N = nblocks * block_size;
+  constexpr int N = nblocks * block_size;
   TypeParam data_contiguous[N];
 
+  constexpr TypeParam value = dlaf_test::TypeUtilities<TypeParam>::element(13, 26);
   for (auto i = 0; i < N; ++i)
-    data_contiguous[i] = dlaf_test::TypeUtilities<TypeParam>::element(value, 0);
+    data_contiguous[i] = value;
 
-  const int root = 0;
-  auto& communicator = world;
-  auto&& message_input = common::make_data(data_contiguous, N);
+  auto&& message_input = common::make_data(static_cast<const TypeParam*>(data_contiguous), N);
   auto&& message_output = common::make_data(data_strided, nblocks, block_size, block_distance);
-  MPI_Op op = MPI_SUM;
 
-  sync::reduce(root, communicator, op, std::move(message_input), std::move(message_output));
-  if (world.rank() == root) {
-    for (std::size_t i_block = 0; i_block < nblocks; ++i_block)
-      for (std::size_t i_element = 0; i_element < block_size; ++i_element) {
-        auto mem_pos = i_block * block_distance + i_element;
-        EXPECT_EQ(value * NUM_MPI_RANKS, data_strided[mem_pos]);
-      }
-  }
-}
-
-TEST_F(ReduceTest, ConstContiguousToStrided) {
-  using TypeParam = int;
-
-  const TypeParam value = 13;
-
-  // 3 blocks, 2 elements each, with a distance of 5 elements between start of each block
-  // E E - - - E E - - - E E    (without padding at the end)
-  const std::size_t nblocks = 3;
-  const std::size_t block_size = 2;
-  const std::size_t block_distance = 5;
-
-  const std::size_t memory_footprint = (nblocks - 1) * block_distance + block_size;
-  TypeParam data_strided[memory_footprint];
-
-  const int N = nblocks * block_size;
-  TypeParam data_contiguous[N];
-  const TypeParam* const_data_contiguous = data_contiguous;
-
-  for (auto i = 0; i < N; ++i)
-    data_contiguous[i] = dlaf_test::TypeUtilities<TypeParam>::element(value, 0);
-
-  const int root = 0;
+  constexpr int root = 0;
   auto& communicator = world;
-  auto&& message_input = common::make_data(const_data_contiguous, N);
-  auto&& message_output = common::make_data(data_strided, nblocks, block_size, block_distance);
   MPI_Op op = MPI_SUM;
-
   sync::reduce(root, communicator, op, std::move(message_input), std::move(message_output));
+
   if (world.rank() == root) {
     for (std::size_t i_block = 0; i_block < nblocks; ++i_block)
       for (std::size_t i_element = 0; i_element < block_size; ++i_element) {
         auto mem_pos = i_block * block_distance + i_element;
-        EXPECT_EQ(value * NUM_MPI_RANKS, data_strided[mem_pos]);
+        EXPECT_EQ(value * static_cast<TypeParam>(NUM_MPI_RANKS), data_strided[mem_pos]);
```

Comment:
Are you casting an `int` into an `int`?
If you want to allow to change easily TypeParam to another type, you should consider that a floating point type can be used.

Reply:
Sorry, I'm not getting your point.

`using TypeParam = std::complex<double>;`

What am I missing?

Reply:
That floating point results of two equivalent math operations may not be binary equivalent.